sdk/src/main/scala/com/horizen/api/http/TransactionBaseApiRoute.scala

```diff
@@ -0,0 +1,174 @@
+import com.horizen.SidechainNodeViewBase
+import com.horizen.api.http.JacksonSupport._
+import com.horizen.api.http.SidechainTransactionActor.ReceivableMessages.BroadcastTransaction
+import com.horizen.api.http.TransactionBaseErrorResponse._
+import com.horizen.api.http.TransactionBaseRestScheme._
+import com.horizen.block.{SidechainBlockBase, SidechainBlockHeaderBase}
+import com.horizen.chain.AbstractFeePaymentsInfo
+import com.horizen.consensus.ForgingStakeInfo
+import com.horizen.node._
+import com.horizen.serialization.Views
+import com.horizen.transaction._
+import com.horizen.utils.BytesUtils
+
+import java.util.{Optional => JOptional}
+import sparkz.core.serialization.SparkzSerializer
+
+import scala.collection.JavaConverters._
+import scala.collection.convert.ImplicitConversions.`iterable AsScalaIterable`
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+abstract class TransactionBaseApiRoute[
+  TX <: Transaction,
+  H <: SidechainBlockHeaderBase,
+  PM <: SidechainBlockBase[TX, H],
+  FPI <: AbstractFeePaymentsInfo,
+  NH <: NodeHistoryBase[TX, H, PM, FPI],
+  NS <: NodeStateBase,
+  NW <: NodeWalletBase,
+  NP <: NodeMemoryPoolBase[TX],
+  NV <: SidechainNodeViewBase[TX, H, PM, FPI, NH, NS, NW, NP]](
+                                  sidechainTransactionActorRef: ActorRef,
+                                  companion: SparkzSerializer[TX])
+                                 (implicit val context: ActorRefFactory, override val ec: ExecutionContext)
+  extends SidechainApiRoute[TX, H, PM, FPI, NH, NS, NW, NP, NV] {
+
+  /**
+    * Returns an array of transaction ids if formatMemPool=false, otherwise a JSONObject for each transaction.
+    */
+  def allTransactions: Route = (post & path("allTransactions")) {
+    entity(as[ReqAllTransactions]) { body =>
+      withNodeView { sidechainNodeView =>
+        val unconfirmedTxs = sidechainNodeView.getNodeMemoryPool.getTransactions()
+        if (body.format.getOrElse(true)) {
+          ApiResponseUtil.toResponse(RespAllTransactions(unconfirmedTxs.asScala.toList))
+        } else {
+          ApiResponseUtil.toResponse(RespAllTransactionIds(unconfirmedTxs.asScala.toList.map(_.id)))
+        }
+      }
+    }
+  }
+
+  /**
+    * Return a JSON representation of a transaction given its byte serialization.
+    */
+  def decodeTransactionBytes: Route = (post & path("decodeTransactionBytes")) {
+    entity(as[ReqDecodeTransactionBytes]) { body =>
+      companion.parseBytesTry(BytesUtils.fromHexString(body.transactionBytes)) match {
+        case Success(tx) =>
+          //TO-DO JSON representation of transaction
+          ApiResponseUtil.toResponse(RespDecodeTransactionBytes(tx))
+        case Failure(exp) =>
+          ApiResponseUtil.toResponse(ErrorByteTransactionParsing(exp.getMessage, JOptional.of(exp)))
+      }
+    }
+  }
+
+  def allActiveForgingStakeInfo: Route = (post & path("allActiveForgingStakeInfo")) {
```

Comment:
I'm not sure that the stake info we return here is "active".
Actually we return the forging stake the current state. But it is not active, only the snapshot of all stakes in the end of the consensus epoch `N` will be active at epoch `N+2`.

These 2 methods can be renamed without word "Active".
Moreover, do we wan't them to be a part of `transaction` API route? We may move it to `block` as other forger related APIs. Or even to `debug` methods. What do you think?

Reply:
After discussion has been decided to remove these methods. Checks can be made by inspecting the forged block header field `forgingStakeInfo` for expected `stakeAmount` value. The python test has been modified accordingly.